lib/sdr_client/cli.rb

```diff
@@ -31,32 +33,19 @@ def self.default_url
 
     desc 'get DRUID', 'Retrieve an object from the SDR'
     def get(druid)
-      say SdrClient::Find.run(druid, url: options[:url], logger: Logger.new($stderr))
-    rescue SdrClient::Credentials::NoCredentialsError
-      say_error 'Log in first'
-      exit(1)
+      rescue_expected_exceptions do
+        say Find.run(druid, url: options[:url], logger: Logger.new($stderr))
+      end
     end
 
-    desc 'login', 'Prompt for email & password and create a login token (saved in ~/.sdr/token)'
+    desc 'login', 'Open authentication proxy UI, or prompt for username and password, and then prompt for token (saved in ~/.sdr/credentials)'
     def login
-      status = SdrClient::Login.run(
-        url: options[:url],
-        login_service: lambda do
-          {
-            email: ask('Email:'),
-            password: ask('Password:', echo: false)
-          }
-        end
-      )
-      return puts unless status.failure?
-
-      say_error status.failure
-      exit(1)
+      authentication_proxy_url ? login_via_proxy : login_via_credentials
```

Comment:
Note: `login_via_credentials` will be used only for sdr-api instances that are not registered in `config/settings.yml` (as the keys for the `authentication_proxy_url` hash), e.g., instances running locally or in containerland.